DigitalLearningSolutions.Data/DataServices/RegistrationDataService.cs

```diff
@@ -14,7 +14,8 @@ public interface IRegistrationDataService
     {
         (int delegateId, string candidateNumber) RegisterNewUserAndDelegateAccount(
             DelegateRegistrationModel delegateRegistrationModel,
-            bool registerJourneyContainsTermsAndConditions
+            bool registerJourneyContainsTermsAndConditions,
+            bool isRegisteredByAdmin

```

Comment:
I think you've got a tendency to pass context information ("On my account page", "registered by admin") as parameters into lower level methods, then have a single low level method that branches to handle different contexts.
As a result:
    * The behaviour for a particular context is spread down the stack of methods rather than being in one place
    * The low level method depends on all the contexts, so may have to change if any of the business logic for any of the contexts changes.
    * Any new scenarios which want to use that low level method might have to modify that low level method to extend the set of context parameters passed into the method (because it might want to do something slightly different from what's currently there) - this runs the risk of breaking existing code that calls that method.
    
A better approach is to make everything depend on abstractions (dependency inversion) - decide on a thing that needs to happen (e.g. your low level method does a thing

Reply:
Oh, apparently this method isn't used outside of RegistrationService either - can you remove it from the interface?

Reply:
Can we rename `isRegisteredByAdmin` to `shouldAssumeEmailVerified`? or something of that ilk. Then we can look at the RegisterByAdmin methods (e.g. `RegisterDelegateByCentre()`) and see that we're automatically verifying the delegate email in that case, rather than having to follow the call stack into `RegistrationDataService` to see what we do if `isRegisteredByAdmin`.

Reply:
Sure! renamed